rust/connlib/tunnel/src/tests.rs

```diff
@@ -1279,3 +1375,14 @@ fn sim_relay_prototype() -> impl Strategy<Value = SimRelay<u64>> {
             buffer: vec![0u8; (1 << 16) - 1],
         })
 }
+
+fn upstream_dns_servers() -> impl Strategy<Value = Vec<DnsServer>> {
+    collection::vec(
+        any::<IpAddr>().prop_map(|ip| DnsServer::from((ip, 53))),
+        0..4,
+    )
+}
+
+fn system_dns_servers() -> impl Strategy<Value = Vec<IpAddr>> {
+    collection::vec(any::<IpAddr>(), 1..4) // Always need at least 1 system DNS server. TODO: Should we test what happens if we don't?
```

Comment:
what should happen without a system DNS server is that cidr resources still work, maybe it's worth testing that?

Reply:
I added it to https://github.com/firezone/firezone/issues/5099.